src/general_tally.cpp

```diff
@@ -206,6 +206,98 @@ void GeneralTally::score_source(const BankedParticle& p) {
   tally_gen_score_.element(indices.begin(), indices.end()) += source_score;
 }
 
+// method to evaluate the tally
+double GeneralTally::evaluate(const Position r, const double E) const {
+  StaticVector4 indices;
+  double inv_dV;
+  // if both energy and position filter don't exist, then index is 0
+  if (position_filter_ == nullptr && energy_in_ == nullptr) {
+    indices.push_back(0);
+  } else {
+    // get the index for the energy if exist
+    if (energy_in_) {
+      std::optional<std::size_t> E_indx = energy_in_->get_index(E);
+      if (E_indx.has_value() == false) {
+        // Not inside any energy bin. Therefore, return 0.
+        return 0.;
+      }
+      indices.push_back(E_indx.value());
+    }
+    // create a temporary tracker
+    Direction u;
+    Tracker trkr(r, u);
+    // get the indices for the positions, if exist
+    StaticVector3 position_indices;
+    if (position_filter_) {
+      // It will provde the reduce dimensions
+      position_indices = position_filter_->get_indices(trkr);
+
+      if (position_indices.empty()) {
+        // Not inside any position bin. Therefore, return 0.
+        return 0.;
+      }
+      indices.insert(indices.end(), position_indices.begin(),
+                     position_indices.end());
+      // get the inverse of the volume of the bin
+      inv_dV = position_filter_->inv_dV(position_indices);
+    }
+  }
+  const double tally_value =
+      inv_dV * tally_avg_.element(indices.begin(), indices.end());
```

Comment:
So we should not be normalizing by the volume here. The reason for this is that it will not always be possible with future position filters. For example, if we have a position filter based on the Cell or Material, we have no way of knowing their volume, and this won't be possible. We should therefore be returning the volume integrated tallies, i.e. V<flx>, and if the user wants <flx> in the region, it is up to them to find V and normalize.